src/main/groovy/pl/allegro/tech/build/axion/release/ConditionalReleaseTask.groovy

```diff
@@ -0,0 +1,27 @@
+package pl.allegro.tech.build.axion.release
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import pl.allegro.tech.build.axion.release.domain.Releaser
+import pl.allegro.tech.build.axion.release.domain.scm.ScmRepository
+import pl.allegro.tech.build.axion.release.infrastructure.di.Context
+
+import java.util.regex.Pattern
+
+class ConditionalReleaseTask  extends DefaultTask {
```

Comment:
i miss some kind of test here - tasks are quite hard to test, so i move all logic into separate workers, thus eliminating all *logic* from tasks (thus releaser). maybe you could do the same?